### ISSUE TITLE
Forced Flask Version <2.1.0 Requirement Is Breaking Build

### ISSUE THREAD

**Issue by dsoprea** [getsentry/sentry-python#1391]
### How do you use Sentry?

Sentry Saas (sentry.io)

### Version

1.5.9

### Steps to Reproduce

Sentry is suddenly requiring that Flask is <2.1.0 as of 1.5.9:

![image](https://user-images.githubusercontent.com/715626/162948084-2d816473-3006-4ca0-aa72-e9f4ac3ba4e2.png)

The changelog doesn't explain it:

![image](https://user-images.githubusercontent.com/715626/162947608-3fe2f794-6d8b-4ef8-bd31-db1ad939ccbd.png)

The PR (https://github.com/getsentry/sentry-python/pull/1379
) doesn't explain it, either. The commit doesn't really explain it other than to say that it's just to not break unit-tests:

(https://github.com/getsentry/sentry-python/commit/9a82f7b8f32a11466da483ddf2172b65cfb07a69

![image](https://user-images.githubusercontent.com/715626/162947692-a1fe6c30-5e37-4a7f-913c-d310dfc33a8c.png)

However, this broke our build and not wanting to break Sentry's unit-tests is not an awesome reason for doing so. As a result, we're having to keep Sentry at 1.5.8 in order to unblock things. Can you explain more why this was necessary? Is there a timeline for getting this figured out so we can go back to using tip? 

Dustin

### Expected Result

No build conflict

### Actual Result

Build conflict:

```
pkg_resources.ContextualVersionConflict: (Flask 2.1.1 (/usr/local/lib/python3.8/dist-packages), Requirement.parse('flask<2.1.0,>=0.11; extra == "flask"'), {'sentry-sdk'})
```

**Comment by antonpirker** [comment]
Hey @dsoprea !
Thanks for your message. 

I think it was because there was a new Werkzeug release and with that the SDK did not work anymore so I had to pin the Flask version to make sure the old Werkzeug would be used. (or something like this.)

Does your project works in SDK 1.5.8 with the newest Flask and Werkzeug?

**Comment by dsoprea** [comment]
Yes, it should be. I'm pinning Sentry to 1.5.8 but everything else should be at tip, and, since it's a Docker build, there should be no environmental entanglements with older versions.

**Comment by antonpirker** [comment]
Ok, I created a PR where the version contraint is removed. Let's see what will happen in the test suite:
https://github.com/getsentry/sentry-python/pull/1395

I remember I did this version constraint because a new version of some lib that the newest Flask uses broke the test suite. But I am not 100% sure what it was, because there where several new versions of libs that are dependencies of Flask that had a new release in the last couple of weeks that somehow manage to break the test suite.

**Comment by dsoprea** [comment]
Thanks for attending to this.

**Comment by antonpirker** [comment]
Hey @dsoprea !
We just released 1.5.10 that fixed this problem!

**Comment by antonpirker** [comment]
Thanks again for reporting this!
If you send me your shipping address to anton [dot] pirker {at} sentry (.) io I can send you a couple of stickers as a small "Thank you".
